Nos/Service/FileStorage/FileStorageAPIClient.swift

```diff
@@ -74,11 +76,34 @@ class NostrBuildAPIClient: FileStorageAPIClient {
     }
     
     /// The URL of the uploaded asset parsed from the API's response.
-    private func assetURL(from responseData: Data) throws -> URL {
-        let response = try decoder.decode(FileStorageUploadResponseJSON.self, from: responseData)
-        guard let urlString = response.nip94Event?.urlString else {
-            throw FileStorageAPIClientError.uploadFailed(response.message)
+    private func assetURL(from responseData: Data, response: HTTPURLResponse?) throws -> URL {
+        let decodedResponse = try decoder.decode(FileStorageUploadResponseJSON.self, from: responseData)
+
+        guard let urlString = decodedResponse.nip94Event?.urlString else {
+            /// Assign an empty string if the response message is nil.
+            let message = decodedResponse.message ?? ""
+
+            /// Checks if the response contains a status code of `413 Payload Too Large`.
+            if let errorCode = response?.statusCode, errorCode == 413 {
+                /// Verify if the error message indicates the file size exceeds the limit.
+                if let regex = try? NSRegularExpression(
+                    pattern: "File size exceeds the limit of (\\d*\\.\\d* [MKGT]B)",
+                    options: []
+                ),
+                let match = regex.firstMatch(
+                    in: message,
+                    options: [],
+                    range: NSRange(location: 0, length: message.utf16.count)
+                ),
+                let range = Range(match.range(at: 1), in: message) {
+                    let fileSizeLimit = String(message[range])
+                    throw FileStorageAPIClientError.fileTooBig(fileSizeLimit)
+                }
```

Comment:
I think this `if let...` statement is missing an `else` block. It might be easier to read and see this if we refactored some of this out into a new `func`. I'd suggest something like `func fileSizeLimit(from message: String) -> String?`. So then here, we could doo something like...

```
if let errorCode = response?.statusCode, errorCode == 413 {
    let  fileSizeLimit = fileSizeLimit(from: message)
    throw FileStorageAPIClientError.fileTooBig(fileSizeLimit)
}
```